### PR TITLE
Mesh and Shader fixes

### LINKED ISSUE #847: parseShaderSource fails to parse with some types of block comments
For example, the following code will not be correctly parsed. While debugging, after the parser has reached the eof  the state mahcine says it is still in `BlockCom1` state. 
```c++
/*
**/
```

### LINKED ISSUE #848: Shader Preivew is not showing parsed code for shaders that has not yet been compiled
**Steps to reproduce**
In an empty network:
1. Add a `Background` processor
2. Open the shader preview for background.frag from the shader menu
3. Press the pound symbol (i.e `#`) to see the preview
4. Result:  
image

### PR DESCRIPTION
**Pull request by petersteneteg**
closes #847
closes #848

### CODE REVIEW

File: modules/opengl/include/modules/opengl/buffer/bufferobject.h [hunk 1]
```diff
@@ -75,6 +78,11 @@ class IVW_MODULE_OPENGL_API BufferObject : public Observable<BufferObjectObserve
     void bind() const;
     void unbind() const;
 
+    void bindBase(GLuint index) const;
```
Comment (ResearchDaniel) [inline modules/opengl/include/modules/opengl/buffer/bufferobject.h:95]: Unclear what this means and how it is different from bind()

File: modules/opengl/include/modules/opengl/buffer/bufferobject.h [hunk 2]
```diff
@@ -75,6 +78,11 @@ class IVW_MODULE_OPENGL_API BufferObject : public Observable<BufferObjectObserve
...
+    void bindRange(GLuint index, GLintptr offset, GLsizeiptr size) const;
```
Comment (ResearchDaniel) [inline modules/opengl/include/modules/opengl/buffer/bufferobject.h]: What is index? See comment above

File: modules/opengl/include/modules/opengl/buffer/bufferobject.h (hunk 1 above)
Comment (ResearchDaniel) [reply modules/opengl/include/modules/opengl/buffer/bufferobject.h:95]: How about 
```
/*
 * \brief Calls glBindBufferBase. 
 * Binds the buffer at bufferTargetIndex of the array of targets 
 * specified by the associated target ( @see getTarget )
 * Targets must be one of GL_ATOMIC_COUNTER_BUFFER, 
 * GL_TRANSFORM_FEEDBACK_BUFFER, GL_UNIFORM_BUFFER or GL_SHADER_STORAGE_BUFFER
 */ 
void bind(GLuint bufferTargetIndex)
```

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]

**Review by r-englund — APPROVED** [review]
Could cherry-pick 2c03b14b6eeea05eb2990dd05cbda4a1940eb191 for updated unittest

**Review by ResearchDaniel — APPROVED** [review]

**Pull request merged**